src/qtism/common/dom/SerializableDomDocument.php

```diff
@@ -24,27 +24,71 @@
 namespace qtism\common\dom;
 
 use DOMDocument;
+use Exception;
 
 /**
  * Serializable DOM Document
  *
  * This class is a PHP Serializable DOMDocument implementation.
+ * @serializable
  */
-class SerializableDomDocument extends DOMDocument
+class SerializableDomDocument
 {
-    private $xmlData;
+    private DOMDocument $dom;
+
+    public function __construct(string $version = '1.0', string $encoding = '')
+    {
+        $this->dom = new DOMDocument($version, $encoding);
+    }
+
+    public function __serialize(): array
+    {
+        return [
+            'version'  => $this->dom->xmlVersion,
+            'encoding' => $this->dom->encoding,
+            'xmlData'  => (string)$this->__toString()
+        ];
+    }
+
+    public function __unserialize(array $data): void
+    {
+        $this->dom = new DOMDocument($data['version'], $data['encoding']);
+        $this->dom->loadXML($data['xmlData']);
+    }
+
+    public function __toString(): string
+    {
+        return $this->dom->saveXML();
+    }
 
     /**
-     * @return array
+     * @throws Exception
      */
-    public function __sleep()
+    public function __call($name, $arguments)
```

Comment:
Adding some `@method` and `@property` annotations describing the ones available in DOMDocument would help. Let's have some on the class level.

Reply:
Done added to class head